assets/js/base/components/cart-checkout/shipping-rates-control-package/package-rates.tsx

```diff
@@ -50,6 +61,25 @@ const PackageRates = ( {
 		}
 	}, [ onSelectRate, rates, selectedOption ] );
 
+	// Set validation errors and display an error notice if there are no shipping rates available.
+	useEffect( () => {
+		if ( rates.length === 0 ) {
+			setValidationErrors( {
+				'shipping-rates-error': SHIPPING_RATE_ERROR,
+			} );
+			createErrorNotice(
+				decodeEntities(
+					'Shipping options are not available. Add a different address for shipping options'
+				),
+				{
+					id: 'shipping-rates-error',
+					context: 'wc/checkout',
+					isDismissible: true,
+				}
+			);
+		}
+	}, [ rates, setValidationErrors, createErrorNotice ] );
```

Comment:
Shouldn't this error trigger at the shipping selector level? ideally we would only run into this scenario when we have both local pickup and regular shipping.

Reply:
Yes, thank you for the suggestion. I have moved the code to the Shipping selector.